CMakeLists.txt

```diff
@@ -186,22 +186,16 @@ target_compile_definitions( freertos_plus_fat
 
 target_compile_options( freertos_plus_fat
   PRIVATE
+    $<$<COMPILE_LANG_AND_ID:C,GNU>:-Wno-array-bounds>
```

Comment:
This is compiling with clang 14.0

Reply:
Nit: I feel array-bounds violations are fairly useful warnings, but I'm unaware whether there were intended "false positive" types being reported. Regardless plenty of other warnings were fixed, so approved first round of reviews.

Reply:
I built POSIX test on Ubuntu 22.04 with GCC11 with `Warray-bounds` and no array bound warnings came up.